plugins/powerpaste.md

```diff
@@ -186,32 +186,94 @@ tinymce.init({
 
 ## Advanced Config Options
 
-### Post filter callback
+### Pre- and post- filtering callbacks
 
-Developers can add customer filtering after **PowerPaste** filters are run using the post filter callback. This can be added as an init option or at runtime by adding the event listener.
+Developers can add custom filtering before and after **PowerPaste's** filters are run using the pre- and post- filtering callbacks. These can be added as init options or at runtime using event listeners.
 
-##### Using the init option
+> Note: These callbacks are also triggered by the core Paste plugin, but when triggered by PowerPaste they are passed more data.
+
+#### Configuration Options
+
+##### paste_preprocess
+
+This setting allows you to run custom filtering on the content from the clipboard before it is run through PowerPaste's filters. The callback is passed two arguments: the PowerPaste plugin instance and an object containing event data. As well as standard event data, this object contains:
+
+- `content` - A string containing the content to be pasted
+- `mode` - A string indicating whether PowerPaste is in `clean` or `merge` mode.
+- `source` - A string indicating which kind of filtering PowerPaste will run based on the source of the content. This will return `html`, `msoffice`, `googledocs`, `image`, `plaintext`, `text` or `invalid`.
+
+Example TinyMCE configuration:
+
+```js
+tinymce.init({
+  selector: "textarea",
+  plugins: "powerpaste",
+  paste_preprocess: function (pluginApi, data) {
+    console.log(data.content, data.mode, data.source);
+    // Apply custom filtering by mutating data.content
+    const content = data.content;
+    const newContent = customFilter(content);
```

Comment:
This could be a bit confusing since it looks like it's calling something that you may not define yourself. Maybe rename this to `yourCustomFilter` or add a comment to make it very clear this should be a function they implement themselves?